testing/integration/logs_ingestion_test.go

```diff
@@ -127,6 +131,37 @@ func startMockES(t *testing.T) string {
 	return s.URL
 }
 
+// checkHealthAtStartup ensures all the beats and agent are healthy and working before we continue
+func checkHealthAtStartup(t *testing.T, ctx context.Context, agentFixture *atesting.Fixture) {
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	compDebugName := ""
+	require.Eventually(t, func() bool {
+		allHealthy := true
+		status, err := agentFixture.ExecStatus(ctx)
+		if err != nil {
+			t.Logf("agent status returned an error: %v", err)
+			return false
+		}
+
+		for _, comp := range status.Components {
+			// make sure the components include the expected integrations
+			for _, v := range comp.Units {
+				t.Logf("unit ID: %s", v.UnitID)
+				t.Logf("unit state: %s", v.Message)
+				if v.State != int(cproto.State_HEALTHY) {
+					allHealthy = false
+				}
+			}
+			t.Logf("component state: %s", comp.Message)
+			if comp.State != int(cproto.State_HEALTHY) {
+				compDebugName = comp.Name
+				allHealthy = false
+			}
+		}
+		return allHealthy
+	}, 3*time.Minute, 15*time.Second, "install never became healthy: components did not return a healthy state: %s", compDebugName)
```

Comment:
Is there a reason not to check more frequently than 15 seconds? I'd rather the test succeed faster if it succeeds,

Reply:
No reason tbh. Let me drop it to a couple of seconds.